apps/shell/tash_security.c

```diff
@@ -23,43 +23,129 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <mbedtls/sha256.h>
+#include <mbedtls/pkcs5.h>
+#include <security/security_crypto.h>
 #include "tash_internal.h"
 
-#define SHA256_HASH_nCHAR  (64) // 32 bytes
-#define SHA256_HASH_nBYTE  (32) // 32 bytes
-#define CONVERT_nCHAR      (sizeof(unsigned long) * 2) // unsigned long for strtoul
+#define CONVERT_nCHAR			(sizeof(unsigned long) * 2) // unsigned long for strtoul
+
+#define PBKDF2_SALT_ENCRYPTED	"\x50\x3B\x3C\x3E\xBF\x0D\x3D\x2C\x9D\x6E\xA2\xF3\x62\xF5\x0F\x9F"
+#define PBKDF2_SALT_LEN			(16)
+#define PBKDF2_KEY_nCHAR		(64) // 32 bytes
+#define PBKDF2_KEY_LEN			(32) // 32 bytes
+#define PBKDF2_ITERATION		(777)
+
+#define AES_BLOCK_SIZE			(16)
+#define AES128_KEY				"ss/1"
+
+
+security_error salt_aes_decrypt(char *input, unsigned int input_size, char *output, unsigned int output_size) 
+{
+	if ((input_size <= 0) || (input == NULL) || (output_size < input_size) || (output == NULL) || (input_size % AES_BLOCK_SIZE)) {
+		return SECURITY_INVALID_INPUT_PARAMS;
+	}
+
+	security_handle hnd;
+	security_data encrypt_data;
+	security_data decrypt_data;
+	security_aes_param aes_param;
+	char *tmp = NULL;
+
+	if (0 != security_init(&hnd)) {
+		printf("security_crypto : security_init failed\n");
+		return SECURITY_ERROR;
+	}
+
+	encrypt_data.data = input;
+	encrypt_data.length = input_size;
+
+	decrypt_data.length = output_size;
+
+	aes_param.mode = AES_CBC_NOPAD;
+	aes_param.iv = NULL;
+	aes_param.iv_len = 0;
+
+	if (0 != crypto_aes_decryption(hnd, &aes_param, AES128_KEY, &encrypt_data, &decrypt_data)) {
```

Comment:
crypto_aes_decryption will return various type of error.
so it should print out type of error, it will be helpful when you do debugging
```
security_error ret = crypto_aes_decryption(...);
if (ret != SECURITY_OK) {
    printf("blahblah ret : %d\n", ret);
    return ERROR;
}
```